### PR TITLE
Slew FFT frequency output

### PR DESCRIPTION
**Pull request by andyp1per**
Slew the frequency output of the FFT to the notch filter to avoid shot noise

### CODE REVIEW

**Comment by andyp1per** [comment]
Flew 6 packs on a 4" yesterday with this. Control was good and no particular noise issues.

File: libraries/AP_GyroFFT/AP_GyroFFT.h [hunk 1]
```diff
@@ -274,7 +282,7 @@ class AP_GyroFFT
     // engine health in tracked peaks
     uint8_t _health;
     // engine health on roll/pitch/yaw
-    Vector3<uint8_t> _rpy_health;
+    Vector3<uint32_t> _rpy_health;
```
Comment (tridge) [inline libraries/AP_GyroFFT/AP_GyroFFT.h:285]: why this change? Should this be Vector3<bool> ?

File: libraries/AP_GyroFFT/AP_GyroFFT.h [hunk 2]
```diff
@@ -222,11 +226,15 @@ class AP_GyroFFT
         // fft engine output rate
         uint32_t _output_cycle_ms;
         // tracked frequency peak
-        Vector3<uint8_t> _tracked_peak;
+        Vector3<uint32_t> _tracked_peak;
```
Comment (tridge) [inline libraries/AP_GyroFFT/AP_GyroFFT.h:229]: should this be a Vector3 of enum FrequencyPeak?

Comment (tridge) [reply libraries/AP_GyroFFT/AP_GyroFFT.h:229]: maybe enum FrequencyPeak needs a :uint8_t to make it uint8_t base type

**Comment by tridge** [comment]
i think we need to generate some debug graphs to show that the slewed freq is nice and smooth
maybe we need a slew filter in libraries/Filter?

File: libraries/AP_GyroFFT/AP_GyroFFT.h (hunk 1 above)
Comment (andyp1per) [reply libraries/AP_GyroFFT/AP_GyroFFT.h:285]: Ah, they should be Vector3ul. I made this change because it should save some flash - there's no good reason to use uint8_t over uint32_t and using it requires that the template methods be defined.

File: libraries/AP_GyroFFT/AP_GyroFFT.h (hunk 2 above)
Comment (andyp1per) [reply libraries/AP_GyroFFT/AP_GyroFFT.h:229]: I could do that, but more flash because more template methods defined?

**Comment by andyp1per** [comment]
@tridge - I did check the output in SITL - is that good enough? I originally looked at adding a slew filter in Filters, but it requires a bunch of config and updates that more naturally fit in the caller.

**Comment by andyp1per** [comment]
I ran the FFT logging at 100Hz and change the harmonic test to a 256-length FFT to slow it down. You can see the slew in action:

image

I have change the Vector's back to uint8_t - it's an extra 100 bytes, but I guess the type is worth it. I have also changed the other Vector integral types to use the predefined names.

File: libraries/AP_GyroFFT/AP_GyroFFT.h (hunk 2 above)
Comment (andyp1per) [reply libraries/AP_GyroFFT/AP_GyroFFT.h]: I've made the base type change but setting it to Vector3<FrequencyPeak> causes other typing issues, so I left it as uint8_t
